### PR TITLE
Add support for directly mounted Grape::API::Instance

### PR DESCRIPTION
**Pull request by tlconnor**
This PR adds support for Grape APIs that are mounted directly as `Grape::API::Instance` instead of `Grape::API`.

Appraisals are also added for Grape 1.3.0.

### CODE REVIEW

File: Appraisals [hunk 1]
```diff
@@ -1,40 +1,26 @@
...
-  end
+case RUBY_VERSION
 
-  if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('2.5.0')
-    appraise "ruby-#{RUBY_VERSION}-rails6" do
-      gem 'rails', '~> 6.0'
-      gem 'grape', '1.2.3'
-      gem 'rails-controller-testing'
-      gem 'sqlite3', '~> 1.4'
+when '2.5.3', '2.6.3' then
```
Comment (Maimer) [inline Appraisals]: I am only seeing this gem used in `ops_app` and `property`, which both use ruby 2.6.3, so support for ruby 2.5.3 can probably be dropped as well at this point. (Unless it is used somewhere else that I am missing)

Comment (Maimer) [reply Appraisals]: Additionally, the `.ruby-version` file still lists `ruby-2.3.3` so that can probably be updated to `ruby-2.6.3`

Comment (Maimer) [reply Appraisals]: Just realized that this is a public gem, so I guess it should support more versions than one's used at Appfolio.

Comment (dvicory) [reply Appraisals]: Yeah, we should support more than just what we use internally.

Comment (tlconnor) [reply Appraisals]: Yeah, as a public repo we need to support more versions. I can update the `.ruby-version` though.

File: lib/declarative_authorization/controller/grape.rb [hunk 2]
```diff
@@ -57,13 +57,18 @@ def logger
...
           def api_class
             if options[:for].respond_to?(:base)
-              options[:for].base # Grape >= 1.2.0 controller
+              # Grape >= 1.2.0 endpoint
+              # Authorization::Controller::Grape can be included into either Grape::API
+              # or Grape::API::Instance, so we need to check both.
+              [
+                options[:for],
+                options[:for].base
+              ].detect { |api| api.respond_to?(:decl_auth_context) }
```
Comment (dvicory) [inline lib/declarative_authorization/controller/grape.rb]: TIL `detect` exists and is basically `find`.

File: lib/declarative_authorization/controller/grape.rb [hunk 3]
```diff
@@ -57,13 +57,18 @@ def logger
...
 
-          protected
-
           def api_class
             if options[:for].respond_to?(:base)
-              options[:for].base # Grape >= 1.2.0 controller
+              # Grape >= 1.2.0 endpoint
+              # Authorization::Controller::Grape can be included into either Grape::API
+              # or Grape::API::Instance, so we need to check both.
+              [
```
Comment (dvicory) [inline lib/declarative_authorization/controller/grape.rb]: Is the order significant? It seems we used to check `options[:for].base` first when it used to be in `ControllerPermission`.

Comment (tlconnor) [reply lib/declarative_authorization/controller/grape.rb]: Its only ever going to be one or the other.

**Comment by tlconnor** [comment]
This CI run tests the changes against Grape 1.2:
https://circleci.com/workflow-run/17543936-7083-4fe7-8624-c7b7772849c1

**Pull request merged**